File: Scripts/laptop/palindrome-detection/javascript/palindrome-detection.js
function isPalindrome(str) {
  return str === str.split("").reverse().join("");
}

var r = 1;

function executeTask(i) {
	var a = isPalindrome("saippuakivikauppias");
	return i + 1;
}

for ( var i = 0; i < 2000000000; ++i) {
	r = executeTask(i + r);
}

File: Scripts/laptop/palindrome-detection/java/palindromeDetection.java

public class palindromeDetection {

	private static volatile int r = 1;

	public static int executeTask(int i) {

		pali("saippuakivikauppias" + i);
		return i;
	}

	public static void main(String[] args) {
		for (int i = 0; i < 2000000000; ++i) {
			r = executeTask(i);
		}
	}

	public static boolean pali(String testMe) {
		StringBuilder sb = new StringBuilder(testMe);
		return testMe.equals(sb.reverse().toString());
	}

}

